Realm/RLMSyncUser.mm

```diff
@@ -69,6 +75,34 @@ PermissionGetCallback RLMWrapPermissionResultsCallback(RLMPermissionResultsBlock
     };
 }
 
+void runBlockForUserContext(const std::shared_ptr<SyncUser>& user, std::function<void(CocoaSyncUserContext&)> block) {
```

Comment:
I think it'd be clearer to have a method that simply returns a `std::shared_ptr<CocoaSyncUserContext>` and then let callers use it, rather than having this function-based pattern.